todo_app/app.py

```diff
@@ -6,19 +6,41 @@
 from todo_app.app_config import Config
 from todo_app.viewmodel import ViewModel
 from todo_app.data.user_role import writer_required, isWriterRole
-from flask import Flask, render_template, request, redirect, url_for, g
+from flask import Flask, render_template, request, redirect, url_for, g, has_request_context
 from oauthlib.oauth2 import WebApplicationClient
 import requests
 from todo_app.data.TodoUser import ToDoUser
 from werkzeug.debug import DebuggedApplication
 import todo_app.data.mongo_constants as constants
 from flask_login import current_user
+from flask.logging import default_handler
+import logging 
+from loggly.handlers import HTTPSHandler
+from logging import Formatter
+from pythonjsonlogger import jsonlogger
 
 def create_app():
+    loggingLevel = ""
+    if os.getenv('LOGGING_LEVEL'):
+        loggingLevel = os.getenv('LOGGING_LEVEL').upper()
+    log_level_info = {'DEBUG': logging.DEBUG, 
+                    'INFO': logging.INFO,
+                    'WARNING': logging.WARNING,
+                    'ERROR': logging.ERROR,
+                    }
+    logging.basicConfig(level=log_level_info.get(loggingLevel, logging.INFO), format=f'%(asctime)s %(levelname)s %(name)s %(threadName)s : %(message)s')
+    httpLog = logging.getLogger('werkzeug')
     app = Flask(__name__)
     app.config.from_object(Config())
-    github_id = os.getenv('GITHUB_CLIENT')
+    github_id = os.getenv('GITHUB_CLIENT_ID')
     github_secret = os.getenv('GITHUB_SECRET')
+    if app.config['LOGGLY_TOKEN'] is not None:
+        handler = HTTPSHandler(f'https://logs-01.loggly.com/inputs/{app.config["LOGGLY_TOKEN"]}/tag/todo-app')
+        jsonFormatter = jsonlogger.JsonFormatter("[%(asctime)s] %(levelname)s in %(module)s: %(message)s")
+        textFormatter = Formatter("[%(asctime)s] %(levelname)s in %(module)s: %(message)s")
+        handler.setFormatter(jsonFormatter)
+        httpLog.addHandler(HTTPSHandler(f'https://logs-01.loggly.com/inputs/{app.config["LOGGLY_TOKEN"]}/tag/todoapp-requests'))
+    app.logger.addHandler(handler)
```

Comment:
This should be inside the if statement, otherwise it crashes if no Loggly token is set

Reply:
Will change in my module 14 which is nearing its completion.